### DISCUSSION TITLE
read spreadsheet without knowing the number of rows in the column

### DISCUSSION THREAD

**Discussion by Juniornewxt** [qax-os/excelize#1341] [Q&A]
hello friends, is it possible to read a worksheet and show the data of each column, without knowing the number of rows that will have in that worksheet?  if yes, do you have an example?

**Comment by xuri** [comment]
Iterator row with row's iterator and get cells of each row, here is an example:

```go
package main

import (
    "fmt"

    "github.com/xuri/excelize/v2"
)

func main() {
    f, err := excelize.OpenFile("Book1.xlsx")
    if err != nil {
        fmt.Println(err)
        return
    }
    defer func() {
        // Close the spreadsheet.
        if err := f.Close(); err != nil {
            fmt.Println(err)
        }
    }()
    rows, err := f.Rows("Sheet1")
    if err != nil {
        fmt.Println(err)
        return
    }
    rowNum := 0
    for rows.Next() {
        rowNum++
        row, err := rows.Columns()
        if err != nil {
            fmt.Println(err)
        }
        fmt.Printf("row %d has %d columns\n", rowNum, len(row))
    }
    if err = rows.Close(); err != nil {
        fmt.Println(err)
    }
}
```

**Comment by Juniornewxt** [comment]
I think I expressed myself badly.
I'll try to give an example:
I suppose you have a spreadsheet with the first row A1, A2, with header.
Already B1, B2, C1, C2 .... infinitely, it will have the values ​​I need, but I will never know if the worksheet turns with B1, B2, C1, C2, D1, D2 ... with data or only B1 and B2 with data, is this possible?

**Comment by Juniornewxt** [comment]
I think I'm still explaining it wrong hahaha.
A1 and B1 = header
A2, B2
A3, B3
A4, B4...
That's where the data will come from, the problem is that the data can come in A5, B5, for example, or not, I don't know if it will be filled in, when it will be filled in, I don't know if I'll receive it, but I need to take it out what comes from A2, B2 down to send to another place, record elsewhere.

**Reply by xuri** [reply]
Sorry, I don't understand what exactly you mean, maybe you can reference issue #1245

**Reply by Juniornewxt** [reply]
My english is very bad.
I read data from a .csv file by importing "encoding/csv"
using the below structure what i want to do is exactly the same thing but with .xlsx file using your solution you understand what i want to do?
exemp:
type arq struct {
		name string
		type string
	}

	records, err := readData("arq.csv")

	if err != nil {
		log.Fatal(err)
	}

	for _, record := range records {
		dados := arq{
			name: record[0],
			type:  record[1],
		}

--------------- arq.csv
name;type;
yuri;H
